visualizer/parseTree.js

```diff
@@ -237,35 +237,14 @@
   }
 
   function couldZoom(currentRootTrace, traceNode) {
-    return !(currentRootTrace === traceNode ||
-           !(traceNode.expr instanceof ohm.pexprs.Apply));
-  }
-
-  function zoomIn(ui, grammar, rootTrace, showFailures, traceNode) {
-    var zoomOutButton = $('#zoomOutButton');
-    zoomOutButton.textContent = UnicodeChars.ANTICLOCKWISE_OPEN_CIRCLE_ARROW;
-    zoomOutButton._trace = traceNode;
-    zoomOutButton.hidden = false;
-
-    zoomOutButton.onclick = function(e) {
-      zoomOutButton.hidden = true;
-      zoomOutButton._trace = undefined;
-      refreshParseTree(ui, grammar, rootTrace, showFailures);
-    };
-    zoomOutButton.onmouseover = function(e) {
-      var zoomState = {zoomTrace: zoomOutButton._trace, previewOnly: true};
-      refreshParseTree(ui, grammar, rootTrace, showFailures, zoomState);
-    };
-    zoomOutButton.onmouseout = function(e) {
-      var zoomState = zoomOutButton._trace && {zoomTrace: zoomOutButton._trace};
-      refreshParseTree(ui, grammar, rootTrace, showFailures, zoomState);
-    };
-
-    refreshParseTree(ui, grammar, rootTrace, showFailures, {zoomTrace: traceNode});
+    return currentRootTrace !== traceNode &&
+           traceNode.succeeded &&
+           !isPrimitive(traceNode.expr) &&
+           traceNode.expr.ruleName !== 'spaces';
   }
 
   function createTraceElement(ui, grammar, rootTrace, traceNode, parent, input,
-    showFailures, optZoomState) {
+    optActionName, optZoomState) {
```

Comment:
Nit: when continuing a line like this, it should be indented an extra two spaces (we can fix in a follow-up commit).